src/tests/data/xof/cshake.vec

```diff
@@ -20,6 +20,17 @@ Salt = 456d61696c205369676e6174757265
 In = 000102030405060708090A0B0C0D0E0F101112131415161718191A1B1C1D1E1F202122232425262728292A2B2C2D2E2F303132333435363738393A3B3C3D3E3F404142434445464748494A4B4C4D4E4F505152535455565758595A5B5C5D5E5F606162636465666768696A6B6C6D6E6F707172737475767778797A7B7C7D7E7F808182838485868788898A8B8C8D8E8F909192939495969798999A9B9C9D9E9FA0A1A2A3A4A5A6A7A8A9AAABACADAEAFB0B1B2B3B4B5B6B7B8B9BABBBCBDBEBFC0C1C2C3C4C5C6C7
 Out = C5221D50E4F822D96A2E8881A961420F294B7B24FE3D2094BAED2C6524CC166B
 
+# cSHAKE instantiation with a Name but no Salt should work
+Name = 426f74616e2d54657374
+In = 000102030405060708090A0B0C0D0E0F101112131415161718191A1B1C1D1E1F202122232425262728292A2B2C2D2E2F303132333435363738393A3B3C3D3E3F404142434445464748494A4B4C4D4E4F505152535455565758595A5B5C5D5E5F606162636465666768696A6B6C6D6E6F707172737475767778797A7B7C7D7E7F808182838485868788898A8B8C8D8E8F909192939495969798999A9B9C9D9E9FA0A1A2A3A4A5A6A7A8A9AAABACADAEAFB0B1B2B3B4B5B6B7B8B9BABBBCBDBEBFC0C1C2C3C4C5C6C7
+Out = 429F4B1FFD82BAAD7EB191BCFD1C9A89486C2C8D
+
+# cSHAKE instantiations without neither a Name nor a Salt should fail
+# (we don't implement the NIST-specified fallback to vanilla SHAKE in that case)
+Fail = cSHAKE-128 cannot accept a salt length of 0
```

Comment:
Conventionally these KAT tests are only used for checking a map from Inputs to Output and while yeah nominally an exception is the "output" here, I think it would be clearer to skip this part of the change and then just have a one-off test that checks cSHAKE rejects in this case.

Reply:
Moved it into a simpler test inside `run_final_tests()`.

Reply:
Thanks